From d5a0020fad9a1bf45a6bb119b88398fe65bc81b7 Mon Sep 17 00:00:00 2001
From: Daniel P. Berrange <berrange@redhat.com>
Date: Thu, 8 Apr 2010 15:37:24 +0100
Subject: [PATCH 01/16] Turn on JSON mode and -netdev usage for RHEL6 binary

The RHEL6 KVM lives in /usr/libexec. We use this as an indication
that we can prematurely turn on JSON & -netdev usage for QEMU

* src/qemu/qemu_conf.c: Turn on JSON mode and -netdev usage for
  RHEL6 binary
---
 src/qemu/qemu_conf.c |   11 +++++++++++
 1 files changed, 11 insertions(+), 0 deletions(-)

diff --git a/src/qemu/qemu_conf.c b/src/qemu/qemu_conf.c
index df57d88..8a1d19f 100644
--- a/src/qemu/qemu_conf.c
+++ b/src/qemu/qemu_conf.c
@@ -1369,6 +1369,17 @@ int qemudExtractVersionInfo(const char *qemu,
                           &version, &is_kvm, &kvm_version) == -1)
         goto cleanup2;
 
+    /*
+     * RHEL-6 specific hack to enable some features that were backported
+     * Only RHEL-6 puts KVM in /usr/libexec, so we hook off that since
+     * version numbers don't place nice with backports
+     */
+    if (STREQ(qemu, "/usr/libexec/qemu-kvm") &&
+        version >= 12000) {
+        flags |= QEMUD_CMD_FLAG_MONITOR_JSON;
+        flags |= QEMUD_CMD_FLAG_NETDEV;
+    }
+
     if (retversion)
         *retversion = version;
     if (retflags)
-- 
1.6.6.1

